pages/treeherder.py

```diff
@@ -13,9 +14,16 @@
 
 class TreeherderPage(Base):
 
+    _clear_all_menu_locator = (By.CSS_SELECTOR, 'ul.dropdown-menu:nth-child(3) > li:nth-child(4) > a:nth-child(1)')
     _job_details_actionbar_locator = (By.ID, 'job-details-actionbar')
     _job_result_status_locator = (By.CSS_SELECTOR, '#result-status-pane > div:nth-child(1) > span')
     _logviewer_button_locator = (By.ID, 'logviewer-btn')
+    _logviewer_pin_job_locator = (By.CSS_SELECTOR, 'a > span.glyphicon.glyphicon-pushpin')
```

Comment:
This is the job details panel rather than the logviewer, but the locator looks fragile. Again, it would be good to anchor it close to the element given the vagueness of the locator. I wonder if you could click the count? It might not work in Selenium if it's not the interactive element, but it might be worth a try: `#job-details-actionbar .pinned-job-count`